src/components/Slider/Slider.jsx

```diff
@@ -0,0 +1,242 @@
+import useMergeRefs from "../../hooks/useMergeRefs";
+import "./Slider.scss";
+import { bem } from "./SliderCommons";
+import { ensureValueDefault } from "./SliderHelpers";
+import { SliderProvider } from "./SliderContext";
+import SliderBase from "./SliderBase/SliderBase";
+import SliderInfix from "./SliderInfix";
+import { createBemBlockHelper } from "../../helpers/bem-helper";
+
+const Slider = forwardRef(
+  (
+    {
+      ariaLabel,
+      ariaLabeledBy,
+      className,
+      classNameBase,
+      color,
+      "data-testid": dataTestId,
+      disabled,
+      id,
+      isRange,
+      max,
+      min,
+      onChange,
+      step,
+      showValue,
+      size,
+      value,
+      valueDefault,
+      valueFormatter,
+      valueText,
+      // ------ Additional subcomponents' props
+      indicateSelection,
+      prefix,
+      postfix
+    },
+    ref
+  ) => {
+    const componentRef = useRef(null);
+    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
+    const consumerBem = createBemBlockHelper(classNameBase, { isConsume: true });
+    const providerProps = {
+      ariaLabel,
+      ariaLabeledBy,
+      classNameBase,
+      color,
+      dataTestId,
+      disabled,
+      isRange,
+      max,
+      min,
+      onChange,
+      showValue,
+      size,
+      step,
+      value,
+      valueDefault: ensureValueDefault({ valueDefault, isRange, min, max }),
+      valueFormatter,
+      valueText,
+      infixOptions: {
+        prefix,
+        postfix,
+        indicateSelection
+      }
+    };
+    return (
+      <SliderProvider {...providerProps}>
+        <div
+          className={bem("", { disabled, "value-shown": showValue }, className)}
+          data-testid={dataTestId}
+          id={id}
+          ref={mergedRef}
+        >
+          <SliderInfix kind="prefix" />
+          <SliderBase className={consumerBem("base")} />
+          <SliderInfix kind="postfix" />
+        </div>
+      </SliderProvider>
+    );
+  }
+);
+
+Slider.sizes = SIZES_BASIC;
+Slider.colors = COLORS_BASIC;
+
+Slider.propTypes = {
+  // ------ SliderBase props
+  /**
+   * Define a string that labels the current element (Slider)
+   */
+  ariaLabel: PropTypes.string,
+  /**
+   * ElementId of Node that have the text needed for labeling the current element (Slider)
+   */
+  ariaLabeledBy: PropTypes.string,
+  /**
+   * Custom `class name` to be added to the component-root-node
+   */
+  className: PropTypes.string,
+  /**
+   * Custom base of class names for add to component-root and internal-components nodes
+   *  - check Dev Tools for found available Custom/Consumer Classes
+   */
+  classNameBase: PropTypes.string,
+  /**
+   * Color Mode (primary/positive/negative) of the component (Slider)
+   */
+  color: PropTypes.oneOf(Object.values(Slider.colors)),
+  /**
+   * Unique TestId - can be used as Selector for integration tests and other needs (tracking, etc)
+   */
+  "data-testid": PropTypes.string,
```

Comment:
let use the same capital and small letters syntax as in the other props: dataTestId

Reply:
We are using data-testid in such a way in all other components.
![Screen Shot 2022-01-05 at 15 07 01](https://user-images.githubusercontent.com/96241324/148224751-63e21eab-d128-485f-b7c0-5befa95ebcf9.png)

And it have good reasons for it:

```
<MyComponent dataTestId="my-comp">
  <div data-testid="my-comp-child1" />
  <SubComponent dataTestId="my-comp-child2" />
</MyComponent>
```
compare to:
```
<MyComponent data-testid="my-comp">
  <div data-testid="my-comp-child1" />
  <SubComponent data-testid="my-comp-child2" />
</MyComponent>
```

Also sometimes you need to enhance children components with props from parent component and not always you know type of child dom-node or react-component

Reply:
so let's keep the prop name like this :)